### PR TITLE
Add API call to get general network statistics

### PR DESCRIPTION
**Pull request by n1bor**
allchannels is now so slow it timesout.

This API call returns general stats on the network that is useful to know.

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala [hunk 1]
```diff
@@ -177,7 +191,30 @@ class EclairImpl(appKit: Kit) extends Eclair {
...
 
+  override def getNetworkInfoResponse()(implicit timeout: Timeout): Future[GetNetworkInfoResponse] = {
+    for {
+      nodeCount <- allNodes().map{_.size}
+      channelCount <- allChannels().map{_.size}
+
+      (updateCount: Long,avgCltvExpiryDelta: Long,avgHtlcMinimumMsat: Long,avgFeeBaseMsat: Long,avgFeeProportionalMillionths: Long,avgHtlcMaximumMsat: Long, mf: FlagCounter, cf:FlagCounter ) <- allUpdates(None).map{
+        updates => updates.foldLeft(Tuple8(0L, 0L, 0L, 0L, 0L, 0L,FlagCounter(), FlagCounter())) {
+          (t, c) =>
+            Tuple8(t._1 + 1,
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: Instead of this hard-to-read `Tuple8` why don't you directly use a `GetNetworkInfoResponse` here?
I think it would be cleaner.

File: eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala [hunk 2]
```diff
@@ -155,6 +158,46 @@ class EclairImplSpec extends TestKit(ActorSystem("mySystem")) with fixture.FunSu
       }
     })
   }
+import scala.collection.mutable.ListBuffer
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala]: Can you please order imports and run the formatter?
If you're using IntelliJ, `ctrl+alt+o` and `ctrl+alt+l`.

File: eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala [hunk 3]
```diff
@@ -155,6 +158,46 @@ class EclairImplSpec extends TestKit(ActorSystem("mySystem")) with fixture.FunSu
...
+  test("Get network Info works") {f =>
```
[the remark this answers is not in the document]
Comment (t-bast) [reply eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala]: This test is quite hard to follow, could you clean it up?
Something like arrange-act-assert pattern with meaningful names for values we assert on would be easier to read.

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala (hunk 1 above)
Comment (n1bor) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: Not sure this is possible. As GetNetworkInfoResponse include results of the earlier futures too.
Could fill with dummy zeros but think that would make more confusing.
Especially if additional calls/futures were added to get more data.

File: eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala [hunk 4]
```diff
@@ -155,6 +158,46 @@ class EclairImplSpec extends TestKit(ActorSystem("mySystem")) with fixture.FunSu
...
+
+    val ann_a = makeNodeAnnouncement(randomKey, "node-A", Color(15, 10, -70), Nil)
+    val eclair = new EclairImpl(kit)
+    val fResp = eclair.getNetworkInfoResponse() // ask updates filtered by 'b'
+    f.router.expectMsg('nodes)
+    f.router.reply(Seq(ann_a))
+    f.router.expectMsg('channels)
+    val channelId_ab = ShortChannelId(420000, 1, 0)
+    val channelId_bc = ShortChannelId(420000, 2, 0)
+    val chan_ab = fr.acinq.eclair.router.BaseRouterSpec.channelAnnouncement(channelId_ab, randomKey, randomKey, randomKey, randomKey)
```
[the remark this answers is not in the document]
Comment (n1bor) [reply eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala]: Am in EclairImpSpec so do need it. Wanted to make clear was coming from another Spec.

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala (hunk 1 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: I would do something like (I slightly renamed the fields, you'll have to adapt):

```
override def getNetworkInfoResponse()(implicit timeout: Timeout): Future[GetNetworkInfoResponse] = {
    for {
      nodeCount <- allNodes().map(_.size)
      channelCount <- allChannels().map(_.size)
      channelUpdates <- allUpdates(None)
      val networkInfo = channelUpdates.foldLeft(GetNetworkInfoResponse(channelCount, nodeCount, channelUpdates.size, 0, MilliSatoshi(0), MilliSatoshi(0), 0, MilliSatoshi(0))) {
        (current, next) => current.copy(
          avgCltvExpiryDelta = current.avgCltvExpiryDelta + next.cltvExpiryDelta,
          avgFeeBaseMsat = current.avgFeeBaseMsat + next.feeBaseMsat
          // etc
        )
      }
    } yield networkInfo.copy(
      avgFeeBaseMsat = networkInfo.avgFeeBaseMsat / networkInfo.updates
      // etc
    )
  }
```

And I would refactor to make dividing all `avg` fields by the `updateCount` be done in a method in the case class or something like that.

**Comment by t-bast** [comment]
@n1bor I merged #1116 which might be useful to you: you could now simply expose that via the API. Would that be enough information for your use-case?
